lib/persistence/mongo/Base.js

```diff
@@ -0,0 +1,115 @@
+var Q = require("q");
+var mongo_promise = require('./client.js');
+var crypto = require('crypto');
+var RecordNotFoundError = require('./RecordNotFound')
+
+module.exports = function(options){
+
+  var name = options['name'];
+  if ( !name ) { throw new Error("name is required"); }
+
+  var model = options['model'];
+  if ( !model ) { throw new Error("model is required"); }
+
+  var mongo_model = require('../backbone_sync.js')(model)
+
+  mongo_model = mongo_model.extend({
+
+    pk: function(){
+      return this.get('_id');
+    },
+
+    idAttribute: "_id",
+
+    create: function(attributes, options){
+
+      if(!this.collection_name){
```

Comment:
The purpose of `collection_name` in mongo seems to be the same as `name` in redis. I don't really care which one we use, but I think you should get rid of one or the other.